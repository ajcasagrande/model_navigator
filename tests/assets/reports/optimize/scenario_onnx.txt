OptimizeEvent.WORKSPACE_INITIALIZED {'path': PosixPath('onnx_workspace')}
OptimizeEvent.OPTIMIZATION_STARTED {}
OptimizeEvent.PIPELINE_STARTED {'name': 'Preprocessing'}
OptimizeEvent.COMMAND_STARTED {'command': 'InferInputMetadata', 'config_key': None, 'runner_name': None}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.COMMAND_STARTED {'command': 'FetchInputModelData', 'config_key': None, 'runner_name': None}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.COMMAND_STARTED {'command': 'InferOutputMetadata', 'config_key': None, 'runner_name': None}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.COMMAND_STARTED {'command': 'FetchOutputModelData', 'config_key': None, 'runner_name': None}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.COMMAND_STARTED {'command': 'CopyModel', 'config_key': 'onnx', 'runner_name': None}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.PIPELINE_FINISHED {}
OptimizeEvent.PIPELINE_STARTED {'name': 'Finding max batch size for fixed shapes based pipelines'}
OptimizeEvent.COMMAND_STARTED {'command': 'FindMaxBatchSize', 'config_key': None, 'runner_name': None}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.PIPELINE_FINISHED {}
OptimizeEvent.PIPELINE_STARTED {'name': 'TensorRT Conversion'}
OptimizeEvent.COMMAND_STARTED {'command': 'ConvertONNX2TRT', 'config_key': 'trt-fp32', 'runner_name': None}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.COMMAND_STARTED {'command': 'ConvertONNX2TRT', 'config_key': 'trt-fp16', 'runner_name': None}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.PIPELINE_FINISHED {}
OptimizeEvent.PIPELINE_STARTED {'name': 'Correctness'}
OptimizeEvent.COMMAND_STARTED {'command': 'Correctness', 'config_key': 'onnx', 'runner_name': 'OnnxCUDA'}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.COMMAND_STARTED {'command': 'Correctness', 'config_key': 'onnx', 'runner_name': 'OnnxTensorRT'}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.COMMAND_STARTED {'command': 'Correctness', 'config_key': 'trt-fp32', 'runner_name': 'TensorRT'}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.COMMAND_STARTED {'command': 'Correctness', 'config_key': 'trt-fp16', 'runner_name': 'TensorRT'}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.PIPELINE_FINISHED {}
OptimizeEvent.PIPELINE_STARTED {'name': 'Performance'}
OptimizeEvent.COMMAND_STARTED {'command': 'Performance', 'config_key': 'onnx', 'runner_name': 'OnnxCUDA'}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.COMMAND_STARTED {'command': 'Performance', 'config_key': 'onnx', 'runner_name': 'OnnxTensorRT'}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.COMMAND_STARTED {'command': 'Performance', 'config_key': 'trt-fp32', 'runner_name': 'TensorRT'}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.COMMAND_STARTED {'command': 'Performance', 'config_key': 'trt-fp16', 'runner_name': 'TensorRT'}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.PIPELINE_FINISHED {}
OptimizeEvent.PIPELINE_STARTED {'name': 'Verify Models'}
OptimizeEvent.COMMAND_STARTED {'command': 'VerifyModel', 'config_key': 'onnx', 'runner_name': 'OnnxCUDA'}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.COMMAND_STARTED {'command': 'VerifyModel', 'config_key': 'onnx', 'runner_name': 'OnnxTensorRT'}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.COMMAND_STARTED {'command': 'VerifyModel', 'config_key': 'trt-fp32', 'runner_name': 'TensorRT'}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.COMMAND_STARTED {'command': 'VerifyModel', 'config_key': 'trt-fp16', 'runner_name': 'TensorRT'}
OptimizeEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
OptimizeEvent.PIPELINE_FINISHED {}
OptimizeEvent.BEST_MODEL_PICKED {'config_key': 'trt-fp16', 'runner_name': 'TensorRT', 'model_path': PosixPath('onnx_workspace/trt-fp16/model.plan')}
OptimizeEvent.OPTIMIZATION_FINISHED {}
