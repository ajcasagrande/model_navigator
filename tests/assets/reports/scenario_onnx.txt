NavigatorEvent.WORKSPACE_INITIALIZED {'path': PosixPath('onnx_workspace')}
NavigatorEvent.OPTIMIZATION_STARTED {}
NavigatorEvent.PIPELINE_STARTED {'name': 'Preprocessing'}
NavigatorEvent.COMMAND_STARTED {'command': 'InferInputMetadata', 'config_key': None, 'runner_name': None}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.COMMAND_STARTED {'command': 'FetchInputModelData', 'config_key': None, 'runner_name': None}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.COMMAND_STARTED {'command': 'InferOutputMetadata', 'config_key': None, 'runner_name': None}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.COMMAND_STARTED {'command': 'FetchOutputModelData', 'config_key': None, 'runner_name': None}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.COMMAND_STARTED {'command': 'CopyModel', 'config_key': 'onnx', 'runner_name': None}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.PIPELINE_FINISHED {}
NavigatorEvent.PIPELINE_STARTED {'name': 'Finding Max Batch Size for TensorRT'}
NavigatorEvent.COMMAND_STARTED {'command': 'FindMaxBatchSize', 'config_key': None, 'runner_name': None}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.PIPELINE_FINISHED {}
NavigatorEvent.PIPELINE_STARTED {'name': 'TensorRT Conversion'}
NavigatorEvent.COMMAND_STARTED {'command': 'ConvertONNX2TRT', 'config_key': 'trt-fp32', 'runner_name': None}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.COMMAND_STARTED {'command': 'ConvertONNX2TRT', 'config_key': 'trt-fp16', 'runner_name': None}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.PIPELINE_FINISHED {}
NavigatorEvent.PIPELINE_STARTED {'name': 'Correctness'}
NavigatorEvent.COMMAND_STARTED {'command': 'Correctness', 'config_key': 'onnx', 'runner_name': 'OnnxCUDA'}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.COMMAND_STARTED {'command': 'Correctness', 'config_key': 'onnx', 'runner_name': 'OnnxTensorRT'}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.COMMAND_STARTED {'command': 'Correctness', 'config_key': 'trt-fp32', 'runner_name': 'TensorRT'}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.COMMAND_STARTED {'command': 'Correctness', 'config_key': 'trt-fp16', 'runner_name': 'TensorRT'}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.PIPELINE_FINISHED {}
NavigatorEvent.PIPELINE_STARTED {'name': 'Performance'}
NavigatorEvent.COMMAND_STARTED {'command': 'Performance', 'config_key': 'onnx', 'runner_name': 'OnnxCUDA'}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.COMMAND_STARTED {'command': 'Performance', 'config_key': 'onnx', 'runner_name': 'OnnxTensorRT'}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.COMMAND_STARTED {'command': 'Performance', 'config_key': 'trt-fp32', 'runner_name': 'TensorRT'}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.COMMAND_STARTED {'command': 'Performance', 'config_key': 'trt-fp16', 'runner_name': 'TensorRT'}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.PIPELINE_FINISHED {}
NavigatorEvent.PIPELINE_STARTED {'name': 'Verify Models'}
NavigatorEvent.COMMAND_STARTED {'command': 'VerifyModel', 'config_key': 'onnx', 'runner_name': 'OnnxCUDA'}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.COMMAND_STARTED {'command': 'VerifyModel', 'config_key': 'onnx', 'runner_name': 'OnnxTensorRT'}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.COMMAND_STARTED {'command': 'VerifyModel', 'config_key': 'trt-fp32', 'runner_name': 'TensorRT'}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.COMMAND_STARTED {'command': 'VerifyModel', 'config_key': 'trt-fp16', 'runner_name': 'TensorRT'}
NavigatorEvent.COMMAND_FINISHED {'status': CommandStatus.OK}
NavigatorEvent.PIPELINE_FINISHED {}
NavigatorEvent.BEST_MODEL_PICKED {'config_key': 'trt-fp16', 'runner_name': 'TensorRT', 'model_path': PosixPath('onnx_workspace/trt-fp16/model.plan')}
NavigatorEvent.OPTIMIZATION_FINISHED {}
