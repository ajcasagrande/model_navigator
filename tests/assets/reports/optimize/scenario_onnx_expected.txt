Optimization started.
Model: Collecting model information ...
Model: Collecting model information OK
Model: Finding max batch size for fixed shapes based pipelines ...
Model: Finding max batch size for fixed shapes based pipelines OK
Model: Building TensorRT engine trt-fp32 from ONNX model ...
Model: Building TensorRT engine trt-fp32 from ONNX model OK
Model: Building TensorRT engine trt-fp16 from ONNX model ...
Model: Building TensorRT engine trt-fp16 from ONNX model OK
Model: Validating model onnx on OnnxCUDA backend ...
Model: Validating model onnx on OnnxCUDA backend OK
Model: Validating model onnx on OnnxTensorRT backend ...
Model: Validating model onnx on OnnxTensorRT backend OK
Model: Validating model trt-fp32 on TensorRT backend ...
Model: Validating model trt-fp32 on TensorRT backend OK
Model: Validating model trt-fp16 on TensorRT backend ...
Model: Validating model trt-fp16 on TensorRT backend OK
Model: Benchmarking model onnx on OnnxCUDA backend ...
Model: Benchmarking model onnx on OnnxCUDA backend OK
Model: Benchmarking model onnx on OnnxTensorRT backend ...
Model: Benchmarking model onnx on OnnxTensorRT backend OK
Model: Benchmarking model trt-fp32 on TensorRT backend ...
Model: Benchmarking model trt-fp32 on TensorRT backend OK
Model: Benchmarking model trt-fp16 on TensorRT backend ...
Model: Benchmarking model trt-fp16 on TensorRT backend OK
Model: Verifying model onnx on OnnxCUDA backend ...
Model: Verifying model onnx on OnnxCUDA backend OK
Model: Verifying model onnx on OnnxTensorRT backend ...
Model: Verifying model onnx on OnnxTensorRT backend OK
Model: Verifying model trt-fp32 on TensorRT backend ...
Model: Verifying model trt-fp32 on TensorRT backend OK
Model: Verifying model trt-fp16 on TensorRT backend ...
Model: Verifying model trt-fp16 on TensorRT backend OK
Optimization finished for the model.
   Optimization result for max throughput and min latency strategy   
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Optimized backend            ┃ Path                               ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ trt-fp16 on TensorRT backend │ onnx_workspace/trt-fp16/model.plan │
└──────────────────────────────┴────────────────────────────────────┘
